collect/collect.go

```diff
@@ -469,23 +463,42 @@ func (i *InMemCollector) processSpan(sp *types.Span) {
 // cache as "kept".
 // It doesn't do any logging and barely touches metrics; this is about as
 // minimal as we can make it.
-func (i *InMemCollector) ProcessSpanImmediately(sp *types.Span, keep bool, sampleRate uint, reason string) {
-	now := time.Now()
-	trace := &types.Trace{
-		APIHost:     sp.APIHost,
-		APIKey:      sp.APIKey,
-		Dataset:     sp.Dataset,
-		TraceID:     sp.TraceID,
-		ArrivalTime: now,
-		SendBy:      now,
-	}
-	// we do want a record of how we disposed of traces in case more come in after we've
-	// turned off stress relief (if stress relief is on we'll keep making the same decisions)
-	i.sampleTraceCache.Record(trace, keep, reason)
+func (i *InMemCollector) ProcessSpanImmediately(sp *types.Span) (processed bool, keep bool) {
+	_, span := otelutil.StartSpanWith(context.Background(), i.Tracer, "collector.ProcessSpanImmediately", "trace_id", sp.TraceID)
```

Comment:
Given that we're trying to shed load here because we're stressed, I'm reluctant to generate a span in this routine since it's being called on every span. Can you convince me that it's worth it?

Reply:
I added here for debugging purpose for stress relief mode. I'm ok to remove it once we are done with testing. Plus, I don't think users will turn on tracing all the time. The overhead should only occur if they have tracing turned on